AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/Storage.smithy

```diff
@@ -0,0 +1,291 @@
+//#- [GetEncryptedActiveBranchKey](#getencryptedactivebranchkey)
+//#- [GetEncryptedBranchKeyVersion](#getencryptedbranchkeyversion)
+//#- [GetEncryptedBeaconKey](#getencryptedbeaconkey)
+//#- [GetKeyStorageInfo](#getkeystorageinfo)
+  operations: [
+    WriteNewEncryptedBranchKey,
+    WriteNewEncryptedBranchKeyVersion,
+    GetEncryptedActiveBranchKey,
+    GetEncryptedBranchKeyVersion,
+    GetEncryptedBeaconKey,
+    GetKeyStorageInfo,
+  ]
+}
+
+@aws.polymorph#reference(resource: KeyStorageInterface)
+structure KeyStorageInterfaceReference {}
+
+@documentation("WriteNewEncryptedBranchKey persists the active item, decrypt only (version) item, and Beacon Key Item of a newly created Branch Key.")
+operation WriteNewEncryptedBranchKey {
+  input: WriteNewEncryptedBranchKeyInput,
+  output: WriteNewEncryptedBranchKeyOutput,
+}
+@documentation("WriteNewEncryptedBranchKeyVersion persists the new active item, decrypt only (version) item of a newly created Branch Key version.")
+operation WriteNewEncryptedBranchKeyVersion {
+  input: WriteNewEncryptedBranchKeyVersionInput,
+  output: WriteNewEncryptedBranchKeyVersionOutput,
+}
+@documentation("Get the ACTIVE branch key for encryption for an existing branch key.")
+operation GetEncryptedActiveBranchKey {
+  input: GetEncryptedActiveBranchKeyInput,
+  output: GetEncryptedActiveBranchKeyOutput,
+}
+@documentation("Gets a specific version for an existing branch key.")
+operation GetEncryptedBranchKeyVersion {
+  input: GetEncryptedBranchKeyVersionInput,
+  output: GetEncryptedBranchKeyVersionOutput,
+}
+@documentation("Get the beacon key associated with an existing branch key.")
+operation GetEncryptedBeaconKey {
+  input: GetEncryptedBeaconKeyInput,
+  output: GetEncryptedBeaconKeyOutput,
+}
+@documentation("Gets information about the underlying storage system.")
+operation GetKeyStorageInfo {
+  input: GetKeyStorageInfoInput,
+  output: GetKeyStorageInfoOutput,
+}
+
+//= aws-encryption-sdk-specification/framework/key-store/key-storage.md#writenewencryptedbranchkey
+//= type=implication
+//# The WriteNewEncryptedBranchKey caller MUST provide:
+//#
+//#- An [EncryptedHierarchicalKey](#encryptedhierarchicalkey) with a [type](#type) of ActiveHierarchicalSymmetricVersion
+//#- An [EncryptedHierarchicalKey](#encryptedhierarchicalkey) with a [type](#type) of HierarchicalSymmetricVersion
+//#- An [EncryptedHierarchicalKey](#encryptedhierarchicalkey) with a [type](#type) of ActiveHierarchicalSymmetricBeacon
+@documentation("
+The information required to atomically write an a new branch key into a key store.
+The identifiers for all keys passed should be the same.
+")
+structure WriteNewEncryptedBranchKeyInput {
+  @required
+  @documentation("
+  The active representation of this branch key.
+  The decrypted form of this key must be the same as the `Version`.
+  ")
+  Active: EncryptedHierarchicalKey,
+  @required
+  @documentation("
+  The decrypt representation of this branch key.
```

Comment:
suggestion/comment: "The" implies singular. Which, at the creation time, is accurate, but is not forever true.  
```suggestion
  A DECRYPT_ONLY representation of this branch key.
```

Reply:
I don't want to push that constant name.